### DISCUSSION TITLE
up-current class on link when URL has hash fragment

### DISCUSSION THREAD

**Discussion by thetutlage** [unpoly/unpoly#277] [Q&A] [answered]
So, I have a situation where the `up-current` class is not added to an anchor tag when the URL has hash fragment. Lemme share a screenshot that may explain it better

![up-current](https://user-images.githubusercontent.com/1706381/131875500-5b381e86-f90d-44dc-8ec8-60b17c2dbc7a.png)

- Unpoly version - 2.3.0

I can use `up-alias` as well, but that has the risk of adding the same class to another anchor. For example:

- Link 1 ( url = /database ) ( up-alias = /database/* )
- Link 2 ( url = /database/query-builder )

Now, link 1 and 2 both will have the active class

**Comment by thetutlage** [comment]
Currently I went with the following hack :/

```ts
class CustomFeedbackURLs extends up.LinkFeedbackURLs {
  isCurrent(url) {
    return super.isCurrent(url.split('#')[0])
  }
}

up.LinkFeedbackURLs = CustomFeedbackURLs
```

**Comment by triskweline** [comment]
I can't seem to reproduce this.

Can you provide minimal [Glitch](https://glitch.com/) project or [CodePen](https://codepen.io) that reproduces the issue?

We also have a Glitch template that loads Unpoly for you: https://glitch.com/edit/#!/empty-unpoly-project

**Reply by thetutlage** [reply]
Sure. Here it is. If I visit this URL https://noisy-quilled-bike.glitch.me/index.html#foobar (notice the hash fragment), I expect the link with `index.html` to have `up-current` class.

Here's the source code https://glitch.com/edit/#!/noisy-quilled-bike

**Comment by triskweline** [comment, marked as answer]
Issue tracked in #284.
